agent/php_execute.h

```diff
@@ -64,6 +64,46 @@ extern void nr_framework_create_metric(TSRMLS_D);
  *           This is necessary to correctly instrument frameworks and libraries
  *           that are preloaded.
  */
+
 extern void nr_php_user_instrumentation_from_opcache(TSRMLS_D);
 
+#if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO /* PHP8+ */
+/*
+ * Purpose : Call the necessary functions needed to instrument a function by
+ *           starting a transaction or segment for a function that has just
+ * started.  This function is registered via the Observer API and will be called
+ * by the zend engine every time a function begins.  The zend engine directly
+ * provides the zend_execute_data which has all details we need to know about
+ * the function. This and nr_php_execute_observer_fcall_end sum to provide all
+the functionality of nr_php_execute and nr_php _execute_enabled and as such will
+use all the helper functions they also used.
```

Comment:
Looks like comment format got a little messed up here
```suggestion
 * the functionality of nr_php_execute and nr_php _execute_enabled and as such will
 * use all the helper functions they also used.
```